runtime-modules/forum/src/lib.rs

```diff
@@ -598,6 +616,9 @@ decl_module! {
                 ERROR_CATEGORY_DOES_NOT_EXIST
             );
 
+            // TODO: fix this:
+            // &who belongs to the lead that originated transaction
+            // moderator_id is separate account with no way to recover its accountId; yet required by runtime's `is_moderator`
```

Comment:
`set_moderator_category(origin, moderator_id, ...)` in this function lead user can set a moderator for a category. There might be a bad implementation. 

Now it first checks if `origin` is provided by lead and then controls `is_moderator(account_id, moderator_id)` which will always return false when `moderator_id` is not belonging to some lead user (because `account_id` belongs to the lead). There is no way to retrieve `account_id` for `moderator_id` now.

Reply:
oh yes, that is a bug, there is no need to check anything about the moderator.